sdks/java/core/src/main/java/org/apache/beam/sdk/io/TFRecordIO.java

```diff
@@ -448,9 +448,9 @@ protected boolean isSplittable() {
       private long startOfRecord;
       private volatile long startOfNextRecord;
       private volatile boolean elementIsPresent;
-      private byte[] currentValue;
-      private ReadableByteChannel inChannel;
-      private TFRecordCodec codec;
+      private @Nullable byte[] currentValue;
```

Comment:
We have inconsistent placement of `@Nullable` - sometimes before the access modifier, sometimes after. Is it an issue?

Reply:
[Google style](https://google.github.io/styleguide/javaguide.html#s4.8.5-annotations), which we tend to follow, indicates

> Annotations applying to a class, method or constructor[, or field] appear immediately after the documentation block
> ...
> There are no specific rules for formatting annotations on parameters, local variables, or types.

I view `@Nullable` as always applying to a type, consistent with Java 8+ pluggable type systems, and more future-proof because it actually makes sense. In Java 7, applying `@Nullable` to a method is a necessary hack implying that the return type was `@Nullable`. You could debate whether it applies to the field or the type of the field. In all these cases, the actual lexemes can be ordered in any way with the same effect. I regret the many times I have written these in the wrong order. Perhaps I will do a global search and replace some day.